v6.1.0 breaks fig.write_image()

Hi, it seems that setting the `ENABLE_KALEIDO_V0_DEPRECATION_WARNINGS = False` broke the `write_image()` method.

https://github.com/plotly/plotly.py/blob/1462f3f403006edb4044493a96b405e9a9ed0c0a/plotly/basedatatypes.py#L3898-L3912

https://github.com/plotly/plotly.py/pull/5177/files#diff-e30d1d28e90c51821f0eeed0db68656f4317c0754deaec93c3365b8a93b4caefR14

regression test

```python
import plotly.graph_objects as go
import numpy as np

def test_write_image(tmp_path):

    np.random.seed(1)

    N = 100
    x = np.random.rand(N)
    y = np.random.rand(N)

    fig = go.Figure()
    fig.add_trace(go.Scatter(
        x=x,
        y=y,
        mode="markers",
    ))

    fig.write_image(tmp_path / "fig1.png")

    assert (tmp_path /"fig1.png").exists()
```

```
E       AssertionError: assert False
E        +  where False = exists()
E        +    where exists = (PosixPath('/tmp/pytest-of-remi-gau/pytest-9/test_write_image0') / 'fig1.png').exists
```

Version tested

|plotly | kaleido  | test |
|-      |-         |-     |
|6.0.1  |0.2.1     | ✅   |
|6.1.0  |0.2.1     | ❌   |
|6.1.0  |1.0.0rc13 | ❌   |

Also note that kaleido extra dependency

https://github.com/plotly/plotly.py/blob/1462f3f403006edb4044493a96b405e9a9ed0c0a/pyproject.toml#L49

will only work if the use ask for `--prerelease`.

Without this, pip install effectively falls back to 6.0.1.

```bash
$ pip install 'plotly[kaleido]' 

Resolved 3 packages in 6ms
Installed 3 packages in 291ms
 + narwhals==1.39.1
 + packaging==25.0
 + plotly==6.0.1
warning: The package `plotly==6.0.1` does not have an extra named `kaleido`
```

```
$ pip install --prerelease 'plotly[kaleido]'

Resolved 8 packages in 10ms
Installed 6 packages in 189ms
 + choreographer==1.0.7
 + kaleido==1.0.0rc13
 + logistro==1.1.0
 + orjson==3.10.18
 + plotly==6.1.0
 + simplejson==3.20.1
```

--- a/plotly/basedatatypes.py
+++ b/plotly/basedatatypes.py
@@ -3908,7 +3908,7 @@ Invalid property path '{key_path_str}' for layout
                 warnings.warn(
                     ENGINE_PARAM_DEPRECATION_MSG, DeprecationWarning, stacklevel=2
                 )
-            return pio.write_image(self, *args, **kwargs)
+        return pio.write_image(self, *args, **kwargs)
 
     # Static helpers
     # --------------

--- a/tests/test_optional/test_kaleido/test_kaleido.py
+++ b/tests/test_optional/test_kaleido/test_kaleido.py
@@ -1,14 +1,17 @@
+import base64
+from contextlib import redirect_stdout
 from io import BytesIO, StringIO
 from pathlib import Path
 import tempfile
-from contextlib import redirect_stdout
-import base64
+from unittest.mock import patch
 
 from pdfrw import PdfReader
 from PIL import Image
+import plotly.graph_objects as go
 import plotly.io as pio
 from plotly.io.kaleido import kaleido_available, kaleido_major
 import pytest
+
 
 fig = {"data": [], "layout": {"title": {"text": "figure title"}}}
 
@@ -160,3 +163,43 @@ def test_defaults():
     finally:
         pio.defaults.default_format = "png"
         assert pio.defaults.default_format == "png"
+
+
+def test_fig_write_image():
+    """Test that fig.write_image() calls the correct underlying Kaleido function."""
+
+    test_fig = go.Figure(fig)
+    test_image_bytes = b"mock image data"
+
+    if kaleido_major() > 0:
+        patch_funcname = "plotly.io._kaleido.kaleido.calc_fig_sync"
+    else:
+        patch_funcname = "plotly.io._kaleido.scope.transform"
+
+    with patch(patch_funcname, return_value=test_image_bytes) as mock_calc_fig:
+        test_fig.write_image("test_path.png")
+
+        # Verify patched function was called once with fig dict as first argument
+        mock_calc_fig.assert_called_once()
+        args, _ = mock_calc_fig.call_args
+        assert args[0] == test_fig.to_dict()
+
+
+def test_fig_to_image():
+    """Test that fig.to_image() calls the correct underlying Kaleido function."""
+
+    test_fig = go.Figure(fig)
+    test_image_bytes = b"mock image data"
+
+    if kaleido_major() > 0:
+        patch_funcname = "plotly.io._kaleido.kaleido.calc_fig_sync"
+    else:
+        patch_funcname = "plotly.io._kaleido.scope.transform"
+
+    with patch(patch_funcname, return_value=test_image_bytes) as mock_calc_fig:
+        test_fig.to_image()
+
+        # Verify patched function was called once with fig dict as first argument
+        mock_calc_fig.assert_called_once()
+        args, _ = mock_calc_fig.call_args
+        assert args[0] == test_fig.to_dict()

